fixbackend/auth/user_repository.py

```diff
@@ -12,82 +12,113 @@
 #  You should have received a copy of the GNU Affero General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from typing import Annotated, Any, AsyncIterator, Dict, Optional
+from typing import Annotated, Any, AsyncIterator, Dict, List, Optional
 from uuid import UUID
 
 from fastapi import Depends
 from fastapi_users.db.base import BaseUserDatabase
 from fastapi_users_db_sqlalchemy import SQLAlchemyUserDatabase
 from fastapi_users_db_sqlalchemy.generics import GUID
+from sqlalchemy import select
 from sqlalchemy.ext.asyncio import AsyncSession
 
 from fixbackend.auth.models import OAuthAccount, User, orm
-from fixbackend.db import AsyncSessionDependency
+from fixbackend.db import AsyncSessionMakerDependency
+from fixbackend.ids import UserId
+from fixbackend.types import AsyncSessionMaker
+from contextlib import asynccontextmanager
 
 
 class UserRepository(BaseUserDatabase[User, UUID]):
     def __init__(
         self,
-        session: AsyncSession,
+        session_maker: AsyncSessionMaker,
     ) -> None:
-        self.db: SQLAlchemyUserDatabase[orm.User, GUID] = SQLAlchemyUserDatabase(session, orm.User, orm.OAuthAccount)
+        self.session_maker = session_maker
+
+    @asynccontextmanager
+    async def user_db(
+        self, session: Optional[AsyncSession] = None
+    ) -> AsyncIterator[SQLAlchemyUserDatabase[orm.User, GUID]]:
+        if session:
+            yield SQLAlchemyUserDatabase(session, orm.User, orm.OAuthAccount)
+        else:
+            async with self.session_maker() as session:
+                yield SQLAlchemyUserDatabase(session, orm.User, orm.OAuthAccount)
```

Comment:
Neat. To not repeat such code everywhere, consider returning just the session, not the DB. In this case, the code could live in a more general place and could be reused.